### PR TITLE
[JENKINS-58450] Bump the incremental branch-source version

### PR DESCRIPTION
**Pull request by v1v**
Caused https://github.com/jenkinsci/github-branch-source-plugin/pull/235

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -83,7 +83,13 @@
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>github-branch-source</artifactId>
-            <version>2.3.6</version>
+            <version>2.5.4-rc860.9c636841daa2</version>
```
Comment (aaronjwhiteside) [inline pom.xml]: is this correct, seems like a release candidate..?

otherwise thanks for the PR!

File: pom.xml [hunk 2]
```diff
@@ -83,7 +83,13 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.commons</groupId>
```
Comment (aaronjwhiteside) [inline pom.xml]: why is this needed?

File: pom.xml (hunk 1 above)
Comment (v1v) [reply pom.xml]: I didn't mention that particular version should be replaced as soon as https://github.com/jenkinsci/github-branch-source-plugin/pull/235 is merged and released. Sorry about the lack of communication

File: pom.xml (hunk 2 above)
Comment (v1v) [reply pom.xml]: A transitive dependency required when bumping the version for `github-branch-source` from `2.3.6` to `2.5.4...`. I can be more accurate so I'll copy and paste the enforce rules I got when bumping those versions later on.

Comment (aaronjwhiteside) [reply pom.xml]: Why isn't it sucked in transitively by `github-branch-source`?

File: pom.xml (hunk 1 above)
Comment (MarkEWaite) [reply pom.xml:86]: Can be set to 2.6.0 now that https://github.com/jenkinsci/github-branch-source-plugin/pull/235/commits/bb2782d966b263bff08db28b7dd0c0dfadb837db has been released.

File: pom.xml (hunk 2 above)
Comment (aaronjwhiteside) [reply pom.xml:90]: bump @v1v , sorry I've not been more responsbile on this project, new kid, new house etc..

just bumping to see if you want to bump to the latest stable release of the github-branch-source dependency?

**Comment by aaronjwhiteside** [comment]
thanks @v1v and @MarkEWaite, have bumped these versions in another commit: https://github.com/jenkinsci/pipeline-github-plugin/commit/85bc6d9ff02c1ae6b997a0a4f058da08d5984583
